src/Vesting.hl

```diff
@@ -0,0 +1,34 @@
+spending vesting
+
+struct Datum {
+    sponsor: PubKeyHash
+    payee: PubKeyHash
+    deadline: Time
+}
+
+enum Redeemer {
+    Cancel
+    Claim
+}
+
+func main(datum: Datum, redeemer: Redeemer, context: ScriptContext) -> Bool {
+    tx: Tx = context.tx;
+    now: Time = tx.time_range.start;
+	assert(now < datum.deadline,"message");
```

Comment:
https://www.hyperion-bt.org/helios-book/lang/builtins/timerange.html#start

Returns the start [Time](https://www.hyperion-bt.org/helios-book/lang/builtins/time.html) of a TimeRange.

Time represents POSIX time in milliseconds (time since 1970/01/01 00:00:00 UTC).